pkg/scheduler/apis/config/v1beta2/defaults.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	"net"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/util/feature"
+	componentbaseconfigv1alpha1 "k8s.io/component-base/config/v1alpha1"
+	"k8s.io/kube-scheduler/config/v1beta2"
+	"k8s.io/kubernetes/pkg/features"
+	"k8s.io/kubernetes/pkg/scheduler/apis/config"
+	"k8s.io/utils/pointer"
+)
+
+var defaultResourceSpec = []v1beta2.ResourceSpec{
+	{Name: string(corev1.ResourceCPU), Weight: 1},
+	{Name: string(corev1.ResourceMemory), Weight: 1},
+}
+
+func addDefaultingFuncs(scheme *runtime.Scheme) error {
+	return RegisterDefaults(scheme)
+}
+
+// SetDefaults_KubeSchedulerConfiguration sets additional defaults
+func SetDefaults_KubeSchedulerConfiguration(obj *v1beta2.KubeSchedulerConfiguration) {
+
+	if obj.Parallelism == nil {
+		obj.Parallelism = pointer.Int32Ptr(16)
+	}
+
+	if len(obj.Profiles) == 0 {
+		obj.Profiles = append(obj.Profiles, v1beta2.KubeSchedulerProfile{})
+	}
+	// Only apply a default scheduler name when there is a single profile.
+	// Validation will ensure that every profile has a non-empty unique name.
+	if len(obj.Profiles) == 1 && obj.Profiles[0].SchedulerName == nil {
+		obj.Profiles[0].SchedulerName = pointer.StringPtr(v1.DefaultSchedulerName)
+	}
+
+	// For Healthz and Metrics bind addresses, we want to check:
+	// 1. If the value is nil, default to 0.0.0.0 and default scheduler port
+	// 2. If there is a value set, attempt to split it. If it's just a port (ie, ":1234"), default to 0.0.0.0 with that port
+	// 3. If splitting the value fails, check if the value is even a valid IP. If so, use that with the default port.
+	// Otherwise leave the address as-is, it will be caught during validation.
+	defaultBindAddress := net.JoinHostPort("0.0.0.0", strconv.Itoa(config.DefaultInsecureSchedulerPort))
+	if obj.HealthzBindAddress == nil {
+		obj.HealthzBindAddress = &defaultBindAddress
+	} else {
+		if host, port, err := net.SplitHostPort(*obj.HealthzBindAddress); err == nil {
+			if len(host) == 0 {
+				host = "0.0.0.0"
+			}
+			hostPort := net.JoinHostPort(host, port)
+			obj.HealthzBindAddress = &hostPort
+		} else {
```

Comment:
No, what we wanted to stop doing was this:

https://github.com/kubernetes/kubernetes/blob/5b46064cae41395cf775cb445ffdaae1261076c7/pkg/scheduler/apis/config/v1beta1/defaults.go#L79-L82

The else clause in this PR is there so that plain IPs are allowed in addition to IP:port. We no longer allow inputs that are invalid IP and invalid IP:port, which is what that issue is talking about.

I'll update the release note and mark the completed items in a bit.

Reply:
ahh, it was the next else. Sorry :)